lnwallet/chancloser/rbf_coop_transitions.go

```diff
@@ -0,0 +1,1109 @@
+func (l *RemoteCloseStart) ProcessEvent(event ProtocolEvent, env *Environment,
+) (*CloseStateTransition, error) {
+
+	switch msg := event.(type) {
+	// If we receive a OfferReceived event, we'll make sure they can
+	// actually pay for the fee. If so, then we'll counter sign and
+	// transition to a terminal state.
+	case *OfferReceivedEvent:
+		// To start, we'll perform some basic validation of the sig
+		// message they've sent.
+		switch {
+		// We'll validate that the remote party actually has enough
+		// fees to pay the closing fees.
+		case !l.RemoteCanPayFees(msg.SigMsg.FeeSatoshis):
+			return nil, fmt.Errorf("%w: %v vs %v",
+				ErrRemoteCannotPay,
+				msg.SigMsg.FeeSatoshis,
+				l.RemoteBalance.ToSatoshis())
+
+		// The sequence they send can't be the max sequence, as that would
+		// prevent RBF.
+		case msg.SigMsg.Sequence > mempool.MaxRBFSequence:
+			return nil, fmt.Errorf("%w: %v", ErrNonFinalSequence,
+				msg.SigMsg.Sequence)
+		}
+
+		// With the basic sanity checks out of the way, we'll now
+		// figure out which signature that we'll attempt to sign
+		// against.
+		var (
+			remoteSig input.Signature
+			noClosee  bool
+		)
+		switch {
+		// If our balance is dust, then we expect the CloserNoClosee
+		// sig to be set.
+		case l.LocalAmtIsDust():
+			if msg.SigMsg.CloserNoClosee.IsNone() {
+				return nil, ErrCloserNoClosee
+			}
+			msg.SigMsg.CloserNoClosee.WhenSomeV(func(s lnwire.Sig) {
+				remoteSig, _ = s.ToSignature()
+				noClosee = true
+			})
+
+		// Otherwise, we'll assume that CloseAndClosee is set.
+		//
+		// TODO(roasbeef): NoCloserClosee, but makes no sense?
+		default:
+			if msg.SigMsg.CloserAndClosee.IsNone() {
+				return nil, ErrCloserAndClosee
+			}
+			msg.SigMsg.CloserAndClosee.WhenSomeV(func(s lnwire.Sig) {
+				remoteSig, _ = s.ToSignature()
+			})
+		}
+
+		chanOpts := []lnwallet.ChanCloseOpt{
+			lnwallet.WithCustomSequence(msg.SigMsg.Sequence),
+		}
+
+		chancloserLog.Infof("responding to close w/ local_addr=%x, "+
+			"remote_addr=%x, fee=%v",
+			l.CloseChannelTerms.LocalDeliveryScript[:],
+			l.CloseChannelTerms.RemoteDeliveryScript[:],
+			msg.SigMsg.FeeSatoshis)
+
+		// Now that we have the remote sig, we'll sign the version they
+		// signed, then attempt to complete the cooperative close
+		// process.
+		//
+		// TODO(roasbeef): need to be able to omit an output when
+		// signing based on the above, as closing opt
+		rawSig, _, _, err := env.CloseSigner.CreateCloseProposal(
+			msg.SigMsg.FeeSatoshis,
+			l.CloseChannelTerms.LocalDeliveryScript,
+			l.CloseChannelTerms.RemoteDeliveryScript,
+			chanOpts...,
+		)
+		if err != nil {
+			return nil, err
+		}
+		wireSig, err := lnwire.NewSigFromSignature(rawSig)
+		if err != nil {
+			return nil, err
+		}
+
+		localSig, err := wireSig.ToSignature()
+		if err != nil {
+			return nil, err
+		}
+
+		// With our signature created, we'll now attempt to finalize
+		// the close process.
+		//
+		// TODO(roasbef); duplication
```

Comment:
I'm not sure where the duplication is, the other callsite here has different parameters